packages/sdk/test/cart/index.test.tsx

```diff
@@ -0,0 +1,138 @@
+import { set } from 'idb-keyval'
+
+import { Cart, CartItem, createCartStore } from '../../src'
+import { waitFor } from '../waitFor'
+
+const item: CartItem = {
+  id: 'test',
+  quantity: 1,
+}
+
+const initialCart: Cart<CartItem> = {
+  id: '',
+  items: [],
+}
+
+const store = createCartStore(initialCart)
+
+beforeEach(async () => {
+  set(`fs::cart`, null)
+
+  store.set(initialCart)
+
+  await waitFor(() => expect(store.read()).toEqual(initialCart))
+})
+
+describe('CartStore', () => {
+  it('uses initial cart state', () => {
+    expect(store.read()).toEqual(initialCart)
+  })
+
+  it('initial cart meta state is set', async () => {
+    expect(store.isEmpty()).toBe(true)
+  })
+})
+
+describe('addItem', () => {
+  it('adds item to the cart', () => {
+    store.addItem(item)
+
+    expect(store.read().items).toHaveLength(1)
+
+    store.addItem(item)
+
+    expect(store.read().items).toHaveLength(1)
+  })
+
+  it('increments existing item quantity in the cart', () => {
+    store.addItem(item)
+    store.addItem(item)
+
+    expect(store.read().items).toHaveLength(1)
+  })
+
+  it('has gifts', () => {
```

Comment:
Don't know if I understood it well but where is the gift-related code?

Reply:
you are right!